File: frontend/src/tools/bank_journal/types/mapping.ts
/** 映射方案类型（对齐后端 app/schemas/mapping.py）。 */

export interface MappingProfileVersion {
  version_no: number;
  bank_template_version_id?: string | null;
  company_journal_template_version_id?: string | null;
  mappings_json?: Record<string, unknown> | null;
  created_by?: string | null;
  /** 展示名（后端联表解析，供直接显示，避免裸 ID）。 */
  created_by_name?: string | null;
  bank_template_version_no?: number | null;
  company_journal_template_version_no?: number | null;
}

export interface MappingProfile {
  id: string;
  company_id: string;
  company_name?: string | null;
  name: string;
  bank_template_id?: string | null;
  company_journal_template_id?: string | null;
  status: string;
  latest_version: MappingProfileVersion;
}

File: backend/app/tools/bank_journal/schemas/mapping.py
from typing import Any

from pydantic import BaseModel


class MappingProfileVersionCreate(BaseModel):
    bank_template_version_id: str | None = None
    company_journal_template_version_id: str | None = None
    mappings_json: dict[str, Any] | None = None
    created_by: str | None = None


class MappingProfileVersionResponse(MappingProfileVersionCreate):
    version_no: int
    # 展示名（联表解析，供前端直接显示，避免裸 ID）
    created_by_name: str | None = None
    bank_template_version_no: int | None = None
    company_journal_template_version_no: int | None = None


class MappingProfileCreate(BaseModel):
    company_id: str
    name: str
    bank_template_id: str | None = None
    company_journal_template_id: str | None = None
    version: MappingProfileVersionCreate


class MappingProfileResponse(BaseModel):
    id: str
    company_id: str
    company_name: str | None = None
    name: str
    bank_template_id: str | None = None
    company_journal_template_id: str | None = None
    status: str
    latest_version: MappingProfileVersionResponse
